lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/TeamUtils.java

```diff
@@ -56,9 +56,32 @@ public static Team copyDboToDto(DBOTeam dbo) throws DatastoreException {
 		} else {
 			dto.setIcon(String.valueOf(dbo.getIcon()));
 		}
+		setMembershipStatus(dbo.getState(), dto);
 		return dto;
 	}
 
+	public static void setMembershipStatus(String state, Team team) throws DatastoreException{
+		if(team.getCanPublicJoin() == null || team.getCanRequestMembership() == null){
```

Comment:
we can move this logic to the enum.  We also unconditionally set the two Booleans from the state enum.
We also need to change all places in the code to use the state enum and no longer use the booleans.